src/framework/mlt_frame.c

```diff
@@ -470,7 +472,7 @@ int mlt_image_format_size( mlt_image_format format, int width, int height, int *
 			return 4;
 		case mlt_image_yuv422p16:
 			if ( bpp ) *bpp = 0;
-			return 4 * height * width ;
+			return height * ALIGN_TO( width * 2, 32 ) + 2 * height * ALIGN_TO( width, 32 ) ;
```

Comment:
This raises again the question about whether to add row alignment to MLT images. It is not necessarily a bad idea, but it needs to be indicated on the frame with a property (e.g. image_alignment) and the new property factored into these calculations. And then it must be done for all image formats to keep things consistent. Next question is: does all existing image processing code for yuv422, rgb24, and rgb24a need to be adjusted to deal with images whose rows are not aligned? On the other hand, we could just add a note to the documentation comment to mlt_image_yuv422p16 that says all planes are aligned to 32 bytes and strides a multiple of 32.